### DISCUSSION TITLE
"st2 login" command is always failed with new user

### DISCUSSION THREAD

**Discussion by Aliskaa** [StackStorm/st2#5577] [Q&A] [answered]
Hello,

I'm following the documentation : https://docs.stackstorm.com/rbac.html#using-rbac-example to create a new user in StackStorm.

But as soon as I want to log in with this new user, I have the folowing lines
1- active RBAC
```
cat files/st2.user.conf
# /etc/st2/st2.user.conf
#
# This file is provided with no custom overrides and should be utilized by end-users to make any
# desired configuration changes. No changes to this file should be checked into the st2-docker repository.
# The order of merging: st2.conf < st2.docker.conf < st2.user.conf

[rbac]
enable = True
backend = default
```

2- create a new user
```
$ adduser rbac_user2
Adding user `rbac_user2' ...
Adding new group `rbac_user2' (1002) ...
Adding new user `rbac_user2' (1002) with group `rbac_user2' ...
Creating home directory `/home/rbac_user2' ...
Copying files from `/etc/skel' ...
Enter new UNIX password:
Retype new UNIX password:
passwd: password updated successfully
Changing the user information for rbac_user2
Enter the new value, or press ENTER for the default
        Full Name []: RBAC User 2
        Room Number []:
        Work Phone []:
        Home Phone []:
        Other []:
Is the information correct? [Y/n] Y
```
3- reload the rbac
```
$ st2-apply-rbac-definitions --config-file /etc/st2/st2.docker.conf
2022-02-11 08:21:29,713 INFO [-] Connecting to database "st2" @ "mongo:27017" as user "None".
2022-02-11 08:21:29,718 INFO [-] Successfully connected to database "st2" @ "mongo:27017" as user "None".
2022-02-11 08:21:29,956 INFO [-] Loading role definitions from "/opt/stackstorm/rbac/roles/"
2022-02-11 08:21:29,960 INFO [-] Loading user role assignments from "/opt/stackstorm/rbac/assignments/"
2022-02-11 08:21:29,964 INFO [-] Loading group to role map definitions from "/opt/stackstorm/rbac/mappings/"
2022-02-11 08:21:29,964 INFO [-] Synchronizing roles...
2022-02-11 08:21:29,972 INFO [-] Roles synchronized (0 created, 2 updated, 0 removed)
2022-02-11 08:21:29,972 INFO [-] Synchronizing users role assignments...
2022-02-11 08:21:29,996 INFO [-] User role assignments synchronized
2022-02-11 08:21:29,996 INFO [-] Synchronizing group to role maps...
2022-02-11 08:21:29,997 INFO [-] Group to role map definitions synchronized.
```
4 -assignment conf for rbac_user2.yaml
```
cat rbac/assignments/rbac_user2.yaml
---
username: "rbac_user2"
description: "Grant test_role role to rbac_user2 user."
enabled: true
roles:
    - "test_role"

```
5- role for test_role.yaml
```
cat rbac/roles/test_role.yaml
---
name: "test_role"
description: "Owner of pack http_error"
enabled: true
permission_grants:
    -
        resource_uid: "webhook:http_error"
        permission_types:
           - "webhook_send"
```
6- verify to login :

```
$ st2 login rbac_user2 -p 'azerty'
ERROR: Failed to log in as rbac_user1: 401 Client Error: Unauthorized
MESSAGE: Invalid or missing credentials for url: http://st2auth:9100/tokens
```

The log in is OK with st2admin.
I'm working with docker stackstorm

```
$ st2 --version
st2 3.5dev (596c60c23), on Python 3.6.9
```

I don't understand :(

**Comment by arm4b** [comment, marked as answer]
Which authentication method did you configure in your st2 instance? See https://docs.stackstorm.com/authentication.html

The documentation you were following with linux user/password refers to [`PAM`](https://github.com/StackStorm/st2-auth-backend-pam):  https://docs.stackstorm.com/authentication.html#auth-backends 
> For sake of this example let us assume that the identity provider is managed by the OS on which StackStorm runs.


If you didn't change the auth method, the default is flat-file https://docs.stackstorm.com/authentication.html#authentication with configuration in the `/etc/st2/htpasswd`.
Check your `st2.conf` first:
https://github.com/StackStorm/st2/blob/4aac99ba8183a590699491d232ff095873e2f5e4/conf/st2.package.conf#L45-L49

---

If that sounds confusing, feel free to propose a PR to improve the docs adding more clarification about that example, - that would be very welcome! Our docs are open-source: https://github.com/StackStorm/st2docs/blob/master/docs/source/rbac.rst

**Reply by Aliskaa** [reply]
Thanks Armad for informations

I'm using flat-file.

For this,
1 - I added a new user in files/htpasswd with the apache tool : htpasswd.
```
htpasswd /root/stackstorm-dev/files/htpasswd toto
New password:
Re-type new password:
Adding password for user toto
```

2 - I recreated mes containers with docker-compose and connected to st2client.

3 - Connect to toto user
```
st2 login toto -p 'azerty'
Logged in as toto
```

My new user is conencted =D

**Comment by shyreneching** [comment]
Hi I am encountering the same problem and user was added when I tried to add the user in the files/htpasswd. However, when I try to create permissions to the user, it seems that it is not being reflected to the user. What could be the problem here?
